crates/network/src/encode.rs

```diff
@@ -0,0 +1,47 @@
+use bytes::{BufMut, BytesMut};
+use rjacraft_protocol::{Encoder, VarInt};
+
+#[derive(Clone, Debug, Default)]
+pub struct PacketEncoder {
+    pub buf: BytesMut,
+}
+
+impl PacketEncoder {
+    pub fn append_frame(&mut self, bytes: &[u8]) -> anyhow::Result<()> {
+        let data_len = VarInt(bytes.len().try_into().unwrap());
```

Comment:
`unwrap()` is invalid here. The buffer comes from an arbitrary source (the function is `pub`). We **must** produce an error for overflow.